src/main/java/com/cozyhome/onlineshop/validation/impl/OptionalFieldBirthdayConstraintValidator.java

```diff
@@ -12,10 +12,13 @@ public void initialize(ValidOptionalFieldBirthday constraintAnnotation) {
 
     @Override
     public boolean isValid(String birthday, ConstraintValidatorContext constraintValidatorContext) {
-        if (birthday != null && !birthday.isEmpty()) {
-            return birthday.matches("^(?:19|20)\\d{2}-(?:0[1-9]|1[0-2])-(?:0[1-9]|[12][0-9]|3[01])$");
+        if (birthday == null) {
+            return false;
+        }
+        if (birthday.isEmpty()) {
+            return true;
         }
```

Comment:
this two if statements a little bit confused, in one case it's true, in another case false

Reply:
is an optional field, an empty string is allowed, but null is not allowed